src/test/ui/rfc-3086-metavar-expr/nested-macro-declarations.rs

```diff
@@ -0,0 +1,28 @@
+// run-pass
+
+#![feature(macro_metavar_expr)]
+
+macro_rules! a {
+    ( $a:ident ) => {
+        macro_rules! $a {
+            ( $$( $b:ident ),* ) => {
+                $$(
+                    macro_rules! $b {
+                        ( $$$$( $c:ident ),* ) => {
+                            $$$$(
+                                fn $c() -> &'static str { stringify!($c) }
+                            ),*
+                        };
```

Comment:
The more nested, the more double dollar signs will be necessary. Not sure if it is correct

Reply:
https://github.com/rust-lang/rfcs/pull/3086#discussion_r589286116